### PR TITLE
Feat/add Custom Xcode MVVM Template

### PR DESCRIPTION
**Pull request by oradwanomar**
Screen Shot 2022-07-01 at 4 39 35 AM

### CODE REVIEW

File: Template/MVVM Files Generator.xctemplate/WithoutXIB/___FILEBASENAME___ViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+//
+extension ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_productName___ViewModelOutput {
+
+}
+
+// MARK: Private Handlers
+//
+extension ___FILEBASENAMEASIDENTIFIER___ {
+
+}
```
Comment (ahmdmhasn) [inline Template/MVVM Files Generator.xctemplate/WithoutXIB/___FILEBASENAME___ViewModel.swift]:
```suggestion
// MARK: Private Handlers
//
private extension ___FILEBASENAMEASIDENTIFIER___ {

}
```

File: Template/MVVM Files Generator.xctemplate/WithoutXIB/___FILEBASENAME___ViewModelType.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+typealias ___FILEBASENAMEASIDENTIFIER___ = ___VARIABLE_productName___ViewModelInput & ___VARIABLE_productName___ViewModelOutput
+
+protocol ___VARIABLE_productName___ViewModelInput {
+  
+}
+
+protocol ___VARIABLE_productName___ViewModelOutput {
+
+}
```
Comment (ahmdmhasn) [inline Template/MVVM Files Generator.xctemplate/WithoutXIB/___FILEBASENAME___ViewModelType.swift]:
```suggestion
/// ___VARIABLE_productName___ Input & Output
///
typealias ___FILEBASENAMEASIDENTIFIER___ = ___VARIABLE_productName___ViewModelInput & ___VARIABLE_productName___ViewModelOutput

/// ___VARIABLE_productName___ ViewModel Input
///
protocol ___VARIABLE_productName___ViewModelInput {
  
}

/// ___VARIABLE_productName___ ViewModel Output
///
protocol ___VARIABLE_productName___ViewModelOutput {

}
```

File: Template/MVVM Files Generator.xctemplate/WithoutXIB/___FILEBASENAME___ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+//MARK: - Ojective C Functions
+extension ___FILEBASENAMEASIDENTIFIER___ {
+    
+    
+}
+
+
+
```
Comment (ahmdmhasn) [inline Template/MVVM Files Generator.xctemplate/WithoutXIB/___FILEBASENAME___ViewController.swift]:
```suggestion
    // MARK: Outlets

    // MARK: Properties
        
    private let viewModel: ___VARIABLE_productName___ViewModelType

    // MARK: Init
        
    init(viewModel: ___VARIABLE_productName___ViewModelType) {
        self.viewModel = viewModel
        super.init(nibName: nil, bundle: nil)
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    // MARK: Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
}

// MARK: - Actions
//
extension ___FILEBASENAMEASIDENTIFIER___ {
    
}

// MARK: - Configurations
//
extension ___FILEBASENAMEASIDENTIFIER___ {
    
}

// MARK: - Private Handlers
//
private extension ___FILEBASENAMEASIDENTIFIER___ {
}
```

File: Template/MVVM Files Generator.xctemplate/WithXIB/___FILEBASENAME___ViewModelType.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+typealias ___FILEBASENAMEASIDENTIFIER___ = ___VARIABLE_productName___ViewModelInput & ___VARIABLE_productName___ViewModelOutput
+
+protocol ___VARIABLE_productName___ViewModelInput {
+  
+}
+
+protocol ___VARIABLE_productName___ViewModelOutput {
+
+}
```
Comment (ahmdmhasn) [inline Template/MVVM Files Generator.xctemplate/WithXIB/___FILEBASENAME___ViewModelType.swift]:
```suggestion

import Foundation

/// ___VARIABLE_productName___ Input & Output
///
typealias ___FILEBASENAMEASIDENTIFIER___ = ___VARIABLE_productName___ViewModelInput & ___VARIABLE_productName___ViewModelOutput

/// ___VARIABLE_productName___ ViewModel Input
///
protocol ___VARIABLE_productName___ViewModelInput {
  
}

/// ___VARIABLE_productName___ ViewModel Output
///
protocol ___VARIABLE_productName___ViewModelOutput {

}
```
```

File: Template/MVVM Files Generator.xctemplate/WithXIB/___FILEBASENAME___ViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+// MARK: ___FILEBASENAMEASIDENTIFIER___Output
+//
+extension ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_productName___ViewModelOutput {
+
+}
+
+// MARK: Private Handlers
+//
+extension ___FILEBASENAMEASIDENTIFIER___ {
```
Comment (ahmdmhasn) [inline Template/MVVM Files Generator.xctemplate/WithXIB/___FILEBASENAME___ViewModel.swift]:
```suggestion
private extension ___FILEBASENAMEASIDENTIFIER___ {
```

File: Template/MVVM Files Generator.xctemplate/WithXIB/___FILEBASENAME___ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+//MARK: - Ojective C Functions
+extension ___FILEBASENAMEASIDENTIFIER___ {
+    
+    
+}
+
+
+
```
Comment (ahmdmhasn) [inline Template/MVVM Files Generator.xctemplate/WithXIB/___FILEBASENAME___ViewController.swift]:
```suggestion
    // MARK: Outlets

    // MARK: Properties
        
    private let viewModel: ___VARIABLE_productName___ViewModelType

    // MARK: Init
        
    init(viewModel: ___VARIABLE_productName___ViewModelType) {
        self.viewModel = viewModel
        super.init(nibName: nil, bundle: nil)
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    // MARK: Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
}

// MARK: - Actions
//
extension ___FILEBASENAMEASIDENTIFIER___ {
    
}

// MARK: - Configurations
//
extension ___FILEBASENAMEASIDENTIFIER___ {
    
}

// MARK: - Private Handlers
//
private extension ___FILEBASENAMEASIDENTIFIER___ {
}
```

**Review by ahmdmhasn — COMMENTED** [review]
Great effort.. Please check the comments ASAP so we can merge and use it

**Comment by ahmdmhasn** [comment]
@oradwanomar Any update?

**Review by claramounir — APPROVED** [review]

**Comment by oradwanomar** [comment]
> @oradwanomar Any update?

> @ahmdmhasn I need one more approval to merge 🥲

**Review by ziadali22 — APPROVED** [review]

**Pull request merged**